realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -26,11 +26,15 @@
 #include "jni_util/java_global_weak_ref.hpp"
 #include "jni_util/java_method.hpp"
 #include "jni_util/java_class.hpp"
+#include "jni_util/java_exception_thrower.hpp"
 
 using namespace realm;
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
+static const char* PK_CONSTRAINT_EXCEPTION_CLASS = "io/realm/exceptions/RealmPrimaryKeyConstraintException";
```

Comment:
If the exception is used by all modules, yes, I think it is a good idea to have all exceptions in a centralized place. But currently, this exception is only used in this file. Let's see what is the best to do in the future maybe when we get more and more exceptions thrown in the JNI.